frontend/src/scenes/cohorts/Cohort.tsx

```diff
@@ -1,199 +1,325 @@
-                    <div>
-                        <h3 className="l3">Matched Users</h3>
-                        <span>List of users that currently match the criteria defined</span>
-                        {cohort.is_calculating ? (
-                            <div className="cohort-recalculating flex-center">
-                                <Spinner size="sm" style={{ marginRight: 4 }} />
-                                We're recalculating who belongs to this cohort. This could take up to a couple of
-                                minutes.
-                            </div>
+                        {cohort.is_static ? (
+                            <Row gutter={24} className="mt ph-ignore-input">
+                                <Col span={24}>
+                                    <Field
+                                        name="csv"
+                                        label={isNewCohort ? 'Upload users' : 'Add users'}
+                                        data-attr="cohort-csv"
+                                    >
+                                        {({ onValueChange }) => (
+                                            <>
+                                                <span className="mb">
+                                                    Upload a CSV file to add users to your cohort. The CSV file only
+                                                    requires a single column with the user’s distinct ID.
+                                                </span>
+                                                <Dragger
+                                                    name="file"
+                                                    multiple={false}
+                                                    fileList={cohort.csv ? [cohort.csv] : []}
+                                                    accept=".csv"
+                                                    showUploadList={false}
+                                                    beforeUpload={(file: UploadFile) => {
+                                                        onValueChange(file)
+                                                        return false
+                                                    }}
+                                                    className="cohort-csv-dragger"
+                                                >
+                                                    {cohort.csv ? (
+                                                        <>
+                                                            <IconUploadFile
+                                                                style={{ fontSize: '3rem', color: 'var(--muted-alt)' }}
+                                                            />
+                                                            <div className="ant-upload-text">
+                                                                {cohort.csv?.name ?? 'File'} uploaded
```

Comment:
The "Uploaded" text is a bit confusing because the file haven't actually uploaded at this point. Maybe, we just write the file name?